### PR TITLE
Request cid from alternateUrl if provided

### PR DESCRIPTION
**Pull request by zhouyx**

### CODE REVIEW

File: src/service/cid-api.js [hunk 1]
```diff
@@ -92,9 +92,20 @@ export class GoogleCidApi {
       if (!token || this.isStatusToken_(token)) {
         this.persistToken_(TokenStatus.RETRIEVING, TIMEOUT);
       }
+
       const url = GOOGLE_API_URL + apiKey;
       return this.fetchCid_(dev().assertString(url), scope, token)
-          .then(this.handleResponse_.bind(this))
+          .then(response => {
+            const cid = this.handleResponse_(response);
+            if (!cid && response['alternateUrl']) {
```
Comment (cramforce) [inline src/service/cid-api.js]: Just checking: The camel case is what the API returns?

File: src/service/cid-api.js [hunk 2]
```diff
@@ -92,9 +92,20 @@ export class GoogleCidApi {
...
+              // If an alternate url is provided, try again with the alternate url
+              // The client is still responsible for appending API keys to the URL.
+              const altUrl = response['alternateUrl'] + apiKey;
```
Comment (cramforce) [inline src/service/cid-api.js]: I think this is missing the `?key=` unless that comes from the server

**Comment by cramforce** [comment]
Please, if possible try to test this against the production API

File: src/service/cid-api.js (hunk 1 above)
Comment (zhouyx) [reply src/service/cid-api.js]: yes

File: src/service/cid-api.js (hunk 2 above)
Comment (zhouyx) [reply src/service/cid-api.js]: fixed.

**Comment by zhouyx** [comment]
Tested against production API. Was able to see received `alternateUrl` and the send out the request to alternative url afterwards. PTAL

File: test/functional/test-cid-api.js [hunk 3]
```diff
@@ -126,6 +126,28 @@ describes.realWin('test-cid-api', {}, env => {
     });
   });
 
+  it('should try alternative url if API provides', () => {
+    fetchJsonStub.onCall(0).returns(Promise.resolve({
+      json: () => {
+        return {
+          alternateUrl: 'https://ampcid.google.uk/v1/publisher:getClientId',
```
Comment (Gregable) [inline test/functional/test-cid-api.js]: nit, i don't think this tld exists. doesn't really matter for your test, but maybe change to .co.uk.

Comment (zhouyx) [reply test/functional/test-cid-api.js]: done : )

**Pull request merged**